test/extended/networking/network_diagnostics.go

```diff
@@ -0,0 +1,248 @@
+package networking
+
+import (
+	"context"
+	v1 "github.com/openshift/api/operator/v1"
+	applyconfigv1 "github.com/openshift/client-go/config/applyconfigurations/config/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	namespace     = "openshift-network-diagnostics"
+	clusterConfig = "cluster"
+	condition     = "NetworkDiagnosticsAvailable"
+	fieldManager  = "network-diagnostics-e2e"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:NetworkDiagnosticsConfig]", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("network-diagnostics")
+
+	g.AfterEach(func(ctx context.Context) {
+		// Reset network diagnostics config
+		netConfigApply := applyconfigv1.Network(clusterConfig).WithSpec(
+			applyconfigv1.NetworkSpec().WithNetworkDiagnostics(nil),
+		)
+		_, err := oc.AdminConfigClient().ConfigV1().Networks().Apply(ctx, netConfigApply,
+			metav1.ApplyOptions{FieldManager: fieldManager, Force: true})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
```

Comment:
Would it make sense to also check this state, either in a separate test or before each test begins?

Reply:
I've added the test to verify the default behaviour.